Documentation/proposals/202212-scrape-config.md

```diff
@@ -0,0 +1,146 @@
+  render Prometheus' configuration to be invalid. They would like to use a CRD to help the team serve themselves in
+  adding scrape configurations.
+* [@auligh in #2787](https://github.com/prometheus-operator/prometheus-operator/issues/2787#issuecomment-539568397)
+  reported needing a specific scrape configuration to scrape services running on the kubernetes nodes but don't have a
+  Service attached to them. This user wants to deploy these scrape configurations alongside the applications rather than
+  in a centralized way with `additionalScrapeConfigs`.
+* [@bgagnon in #2787](https://github.com/prometheus-operator/prometheus-operator/issues/2787#issuecomment-545510764)
+  mentions that `additionalScrapeConfigs`'s usage is error-prone as with usage, multiple unrelated scrape configurations
+  are bundled together.
+
+Furthermore, currently, there is a lot of code duplication due to the operator supporting several CRDs that generate
+scrape configurations. With the new `ScrapeConfig` CRD, it would be possible to consolidate some of that logic, where
+the other `*Monitor` CRDs could be migrated so that they create a ScrapeConfig resource that would ultimately be used by
+the operator to generate scrape configuration.
+
+## Pitfalls of the current solution
+
+Using `additionalScrapeConfig` comes with drawbacks:
+
+* Teams have to build an infrastructure to add scrape rules in a centralized manner, which creates a bottleneck since a
+  single team becomes responsible for the configuration
+* There is no input validation, which can lead to an invalid prometheus configuration
+
+# Goals
+
+* Provide a way for users to self-service adding scrape targets
+* Consolidate the scrape configuration generation logic in a central point for other resources to use
+
+## Audience
+
+* Users who serve Prometheus as a service and want to have their customers autonomous in defining scrape configs
+* Users who want to manage scrape configs the same way as for services running within the Kubernetes cluster
+* Users who want a supported Kubernetes way of scraping targets outside the Kubernetes cluster
+
+# Non-Goals
+
+* This proposal doesn't aim at covering all the fields in
+  [`<scrape_config>`](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config).
+  Specifically, no service discovery other than `static_configs` or `file_sd_configs` should be implemented at first.
+* Refactoring of the other CRDs is not in scope for the first version
+
+# How
+
+As described by
+[@aulig in #2787](https://github.com/prometheus-operator/prometheus-operator/issues/2787#issuecomment-559776221), we
+will create a new `ScrapeConfig` CRD, this config will act the same as the other CRDs and append scrape configurations
+to the configuration. `ScrapeConfig` will allow for any scraping configuration, while the other CRDs provide sane
+defaults. This will allow for isolated testing of the new `ScrapeConfig` CRD.
```

Comment:
@ArthurSens I would image the best place to add that information is here. Currently we say that the `ScrapeConfig` will act the same as the others and append scrape configurations to the configuration. Should I rephrase to the following?
> [...] we will create a new `ScrapeConfig` CRD, this new CRD will act the same as the other CRDs and append scrape configurations to the configuration. Usage of `ScrapeConfig` doesn't exclude the use of the other CRDs, they are not mutually exclusive.